src/checkers/inference/SlotManager.java

```diff
@@ -84,6 +85,21 @@
      */
     CombVariableSlot createCombVariableSlot(Slot receiver, Slot declared);
 
+    /**
+     * Creates new LubVariableSlot using left slot and right slot, and returns
+     * reference to it if no LubVariableSlot representing least upper bound of
+     * left slot and right slot exists. Otherwise, returns the existing CombVariableSlot.
+     * Left slot and right slot can uniquely identify a LubVariableSlot
+     *
+     * @param left
+     *            left side of least upper bound operation
+     * @param right
+     *            right side of least upper bound operation
+     * @return LubVariableSlot that represents the least upper bound result
+     *         of left slot and right slot
+     */
+    LubVariableSlot createLubVariableSlot(Slot left, Slot right);
```

Comment:
Can LubSlots be uniquely identified by their annotation location alone?

Reply:
I'm not sure whether location can uniquely determine a `LubVariableSlot`, but I think it should. However, the current (and only one) client is `InferenceQualifierHierarchy#leastUpperBound()`, in which annotation location is unknown.

Reply:
Keep this.